app/celery/provider_tasks.py

```diff
@@ -0,0 +1,109 @@
+from app.clients.sms import SmsClientException
+from app.dao.notifications_dao import (
+    update_provider_stats,
+    get_notification_by_id,
+    dao_update_notification, update_notification_status_by_id)
+from app.dao.provider_details_dao import get_provider_details_by_notification_type
+from app.dao.services_dao import dao_fetch_service_by_id
+from app.celery.research_mode_tasks import send_sms_response
+
+from notifications_utils.recipients import (
+    validate_and_format_phone_number
+)
+
+from app.dao.templates_dao import dao_get_template_by_id
+from notifications_utils.template import (
+    Template,
+    unlink_govuk_escaped
+)
+
+
+def retry_iteration_to_delay(retry):
+    """
+    Given current retry calculate some delay before retrying
+    Delay calculated as (1 + retry number) squared * 60
+    0: 60 seconds (1 minute)
+    1: 240 seconds (4 minutes)
+    2: 540 seconds (9 minutes)
+    3: 960 seconds (16 minutes)
+    4: 1500 seconds (25 minutes)
+    :param retry (zero indexed):
+    :return length to retry in seconds:
+    """
+    return ((retry + 1) ** 2) * 60
+
+
+@notify_celery.task(bind=True, name="send-sms-to-provider", max_retries=5, default_retry_delay=5)
+def send_sms_to_provider(self, service_id, notification_id, encrypted_notification):
+    task_start = monotonic()
+
+    service = dao_fetch_service_by_id(service_id)
+    provider = provider_to_use('sms', notification_id)
+    notification = get_notification_by_id(notification_id)
+
+    notification_json = encryption.decrypt(encrypted_notification)
+
+    template = Template(
+        dao_get_template_by_id(notification.template_id, notification.template_version).__dict__,
+        values=notification_json.get('personalisation', {}),
+        prefix=service.name
+    )
+    try:
+        if service.research_mode:
+            send_sms_response.apply_async(
+                (provider.get_name(), str(notification_id), notification_json['to']), queue='research-mode'
+            )
+        else:
+            provider.send_sms(
+                to=validate_and_format_phone_number(notification_json['to']),
+                content=template.replaced,
+                reference=str(notification_id)
+            )
+
+            update_provider_stats(
+                notification_id,
+                'sms',
+                provider.get_name(),
+                content_char_count=template.replaced_content_count
+            )
+
+        notification.sent_at = datetime.utcnow()
+        notification.sent_by = provider.get_name(),
+        notification.content_char_count = template.replaced_content_count
+        dao_update_notification(notification)
+    except SmsClientException as e:
+        try:
+            current_app.logger.error(
+                "SMS notification {} failed".format(notification_id)
+            )
+            current_app.logger.exception(e)
+            self.retry(queue="retry", countdown=retry_iteration_to_delay(self.request.retries))
```

Comment:
This is putting more than one task in the retry queue each time it retries. I think I am getting the same number of tasks in the queue as the number of seconds it will retry in.